### PR TITLE
Log4cxx 0.10.0

### PR DESCRIPTION
**Pull request by extrowerk**

### CODE REVIEW

File: dev-libs/log4cxx/log4cxx-0.10.0~git.recipe [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+notification levels."
+HOMEPAGE="https://logging.apache.org/log4cxx/latest_stable/index.html"
+COPYRIGHT="2004-2007 The Apache Software Foundation"
+LICENSE="Apache v2"
+REVISION="1"
+gitSrcRevision="5f825186936a1876f92b88b371334ff26e997287"
+SOURCE_URI="https://github.com/apache/logging-log4cxx/archive/$gitSrcRevision.zip"
+CHECKSUM_SHA256="c6ab2b7d124fdf62d66d3aeda74e1559d71b3363fbe4520a134f1574b5663115"
+SOURCE_FILENAME="log4cxx-$gitSrcRevision.zip"
+SOURCE_DIR="logging-log4cxx-$gitSrcRevision"
```
Comment (fbrosson) [inline dev-libs/log4cxx/log4cxx-0.10.0~git.recipe]: Suggestion:
```diff
-gitSrcRevision="5f825186936a1876f92b88b371334ff26e997287"
+srcGitRev="5f825186936a1876f92b88b371334ff26e997287"
-SOURCE_URI="https://github.com/apache/logging-log4cxx/archive/$gitSrcRevision.zip"
+SOURCE_URI="https://github.com/apache/logging-log4cxx/archive/$srcGitRev.tar.gz"
-CHECKSUM_SHA256="c6ab2b7d124fdf62d66d3aeda74e1559d71b3363fbe4520a134f1574b5663115"
+CHECKSUM_SHA256="763b07c7a487ff2c42d0e82f7c3fd598083d08dc9f5d8652990295a87d99ddb2"
-SOURCE_FILENAME="log4cxx-$gitSrcRevision.zip"
+SOURCE_FILENAME="log4cxx-$portVersion-$srcGitRev.tar.gz"
-SOURCE_DIR="logging-log4cxx-$gitSrcRevision"
+SOURCE_DIR="logging-log4cxx-$srcGitRev"
```
Advantages:
* Most recipes which use a git snapshot define `srcGitRev`.
* The `.tar.gz` is smaller than the zip archive.
* Including `$portVersion-` in `SOURCE_FILENAME` is slightly better (imo) than having only the `$srcGitRev`.

([HaikuPorter-Guidelines#ordering](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering))

File: dev-libs/log4cxx/log4cxx-0.10.0~git.recipe [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+	cmd:libtoolize$secondaryArchSuffix
+	cmd:make
+	"
+
+defineDebugInfoPackage log4cxx$secondaryArchSuffix \
+	$libDir/liblog4cxx.so.$libVersion
+
+BUILD()
+{
+	chmod +x autogen.sh
```
Comment (fbrosson) [inline dev-libs/log4cxx/log4cxx-0.10.0~git.recipe]:
```diff
- 	chmod +x autogen.sh
```
(You can drop this line if you switch `SOURCE_FILENAME` to `.tar.gz`.)

**Comment by alaviss** [comment]
I think you should change the version to `0.10.0_20180322`. The use of the `~` means that the version is older than `0.10.0`, which it certainly isn't.

File: dev-libs/log4cxx/log4cxx-0.10.0~git.recipe [hunk 3]
```diff
@@ -6,12 +6,12 @@ notification levels."
 HOMEPAGE="https://logging.apache.org/log4cxx/latest_stable/index.html"
 COPYRIGHT="2004-2007 The Apache Software Foundation"
 LICENSE="Apache v2"
+srcGitRev="5f825186936a1876f92b88b371334ff26e997287"
 REVISION="1"
```
Comment (fbrosson) [inline dev-libs/log4cxx/log4cxx-0.10.0~git.recipe]: Oops, you moved `REVISION` after `srcGitRev`...
```diff
+REVISION="1"
 srcGitRev="5f825186936a1876f92b88b371334ff26e997287"
-REVISION="1"
```
BTW, thanks for the switch to `srcGitRev` :)

Comment (extrowerk) [reply dev-libs/log4cxx/log4cxx-0.10.0~git.recipe]: But you told me i have to follow this: https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#ordering

Comment (fbrosson) [reply dev-libs/log4cxx/log4cxx-0.10.0~git.recipe]: Oops, I'm so sorry! That's my fault. I had not noticed that the wiki was wrong!!!

Comment (fbrosson) [reply dev-libs/log4cxx/log4cxx-0.10.0~git.recipe]: I've just [fixed the wiki](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines/_compare/80e5f61%5E...80e5f61).

BTW, without your comment I would not have noticed that the wiki was wrong, so thanks for the heads-up :)

**Pull request merged**